python/cuspatial/cuspatial/geometry/geodataframe.py

```diff
@@ -90,9 +90,30 @@ def groupby(self, *args, **kwargs):
                 result.obj.drop(col, axis=1, inplace=True)
         return result
 
+    # @classmethod
+    # def _from_data(cls, data, index = None, columns = None, ):
+    #     return cudf.DataFrame._from_data(data, index, columns)
+
+    @classmethod
+    def _from_data(
+        cls,
+        data,
+        index = None,
+        columns = None,
+    ):
+        out = cls.__new__(cls)
+        out._data = data
+        if index is None:
+            index = cudf.Index(range(data.nrows))
+        out._index = index
+        if columns is not None:
+            out.columns = columns
+        return out
```

Comment:
This seems identical to `DataFrame._from_data`, so maybe we just not override it at all?

Reply:
Looking at the constructor of this class, it seems like this method probably needs some logic like 
```
if not isinstance(next(iter(data.values())), GeoColumn):
    # Convert cudf Columns to GeoColumns in `data`
return super()._from_data(data, index, columns)
```
unless this method is never called with a dict of `GeoColumn`s.